src/transaction.js

```diff
@@ -51,13 +68,24 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
   var tx = new Transaction()
   tx.version = readUInt32()
 
+  var marker = buffer.readUInt8(offset)
+  var flag = buffer.readUInt8(offset + 1)
+
+  var hasWitnesses = false
+  if (marker === Transaction.ADVANCED_TRANSACTION_MARKER &&
+      flag === Transaction.ADVANCED_TRANSACTION_FLAG) {
+    offset += 2
+    hasWitnesses = true
+  }
+
   var vinLen = readVarInt()
   for (var i = 0; i < vinLen; ++i) {
     tx.ins.push({
       hash: readSlice(32),
       index: readUInt32(),
       script: readScript(),
-      sequence: readUInt32()
+      sequence: readUInt32(),
+      witness: EMPTY_SCRIPT
```

Comment:
Witness data is meant to be interpreted as providing the script stack for the witness program. https://github.com/bitcoin/bitcoin/blob/fc4f4547b7f6ea13d273ad926738fcdffee36295/src/script/interpreter.cpp#L1376

Storing it as an array of Buffers makes parsing signatures super easy! Your functions can now operate on an array (just like classifyOutput does), except the origin of the data can now be witness data, or a decompiled scriptSig.

A stack has multiple elements, and the serialization depends on whether it's segwit or not. The lowest common denominator between a scriptSig and witnessData is that they both decompile to an array of buffers.